### PR TITLE
Add documentation about enabling encryption

### PR DESCRIPTION
**Pull request by FelisiaM**
[#179622297](https://www.pivotaltracker.com/story/show/179622297)

Which other branches should this be merged with (if any)?
master

### CODE REVIEW

**Comment by jncd** [comment]
Thanks @FelisiaM

**Comment by jncd** [comment]
@FelisiaM Any chance you might have a screenshot to go with this augmented procedure? (or the creds to an environment where I can take a snap myself?) 

I'll do some work with the formatting and American-ize the language and get back to you with questions.

**Comment by FelisiaM** [comment]
Here are a few screenshots - I will send you the details for the environment as well.

Screenshot 2021-09-22 at 10 09 36

Screenshot 2021-09-22 at 10 09 48

File: installing-with-azure.html.md.erb [hunk 1]
```diff
@@ -88,13 +88,30 @@ To configure <%= vars.product_short %> with a database:
 
 1. Enter the credentials for the state database.
 
-3. Click **Save**.
+1. Set up encryption for the sensitive data in the database.
 
-3. Return to the <%= vars.ops_manager %> Installation Dashboard.
+    * If you want the sensitive data to be encrypted in the database tick **Enable encryption of the Cloud Service Broker database** and add an encryption password in the list of **Database Encryption Passwords**.
+    For the password configure these fields:
+        * **Label**: Enter a unique password label. Do not change once saved.
```
[the remark this answers is not in the document]
Comment (pivotal-marcela-campo) [reply installing-with-azure.html.md.erb:95]: Same for the "password" field

[the remark this answers is not in the document]
Comment (jncd) [reply installing-with-azure.html.md.erb:95]: Changed! Thanks @pivotal-marcela-campo

**Comment by jncd** [comment]
@FelisiaM and @pivotal-marcela-campo 
Thanks for screenshots and the environment access. 

Please check out the new version: [Configure a State Database](https://docs-pcf-staging.sc2-04-pcf1-apps.oc.vmware.com/csb-azure/1-n/installing-with-azure.html#state-db) and the bonus procedures [Rotate Password on the State Database](https://docs-pcf-staging.sc2-04-pcf1-apps.oc.vmware.com/csb-azure/1-n/installing-with-azure.html#rotate-password) and [Remove Encryption from the State Database](https://docs-pcf-staging.sc2-04-pcf1-apps.oc.vmware.com/csb-azure/1-n/installing-with-azure.html#remove-encryption) at the bottom. 
(Rotating and removing encryption didn't seem to fit with the first-time configuring procedure.)

Let me know if you're available to meet up for quick session, maybe Thursday afternoon (BST). I'd love to add a little more context. I have questions like:
+ Should we mention the TLS Enforcement radio buttons? 
+ "If you want the _sensitive_ data to be encrypted," Wording makes it sound like two kinds of data are in the db: sensitive and insensitive — true? or should we ~~sensitive~~? What kind of data is stored in the small state database? The UI says secrets, maybe that's better?
+ Does anyone need to know/care what kind of encryption it is? Are the "passwords" used as encryption keys? 
+ Is this database small enough that there are no worries about performance with encryption? In general, is this something all users should do in production?

**Comment by FelisiaM** [comment]
Hi Jane,

Happy to chat more in a call - but in the meantime here are some answers.

> Should we mention the TLS Enforcement radio buttons?

I think we should, but we don't have much context other then you can enable/disable and add custom certificates.

> "If you want the sensitive data to be encrypted," Wording makes it sound like two kinds of data are in the db: sensitive and insensitive — true? or should we sensitive? What kind of data is stored in the small state database? The UI says secrets, maybe that's better?

Enabling encryption in this case does not encrypt all the data in the database, but only sensitive data such as IaaS credentials needed during some operations. I would pick the word `sensitive` rather then `secret` and I will make the relevant change in the tile wording as well (will send you a screenshot with the new wording).

> Does anyone need to know/care what kind of encryption it is? 

That's a great question. I am unsure how much of this needs to be documented- is there a pattern of how much other components document around this? Maybe we can follow their suit.

> Are the "passwords" used as encryption keys?

Yes, passwords are converted to encryption keys by combining it with a salt and using PDKDF2 algorithm.

> Is this database small enough that there are no worries about performance with encryption?

We don't expect that encryption will have noticeable performance issues. Maybe it is worth mentioning that during first time setup and key rotation all sensitive data will be re-encrypted with the new key and this will increase the start-up time in that scenario.

> In general, is this something all users should do in production?

We would recommend all users enable encryption in production.

**Comment by jncd** [comment]
Hi @FelisiaM 
Oh wow, thanks for all these answers. 

I think I'll fold this info into the doc and then see how it looks before we talk (if we do). 
I was also thinking I should change "Rotate Password on the State Database" to "Rotate the Encryption Password on the State Database" just to make it obvious that we're not talking about changing the access credentials to the database.

**Comment by jncd** [comment]
_TLS enforcement:_ I'm guessing that more fields appear if I choose the custom button... Is there anything there that needs explanation? 
I would have looked for myself but yesterday's environment is no more... so ephemeral 😉

**Pull request merged**